From 36f2942dfd91c72e44aad6b8db6e3d0e1606aaf6 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Tue, 11 Feb 2025 17:43:45 -0800
Subject: [PATCH] Use proper include path for cholmod.h.

It appears that the `SuiteSparse::CHOLMOD` CMake library advertises its include directory as `$PREFIX/include/suitesparse`, making the proper way to include it `#include <cholmod.h>`. When installed in the system `/usr` prefix, the `suitesparse/cholmod.h` path works by accident. When using a custom built suitesparse at a custom location, this is not so.

I believe that this change is the correct way to do it for all configurations, based on how the SuiteSparse library is set up.
---
 library/src/amd_detail/dlopen/cholmod.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/src/amd_detail/dlopen/cholmod.hpp b/library/src/amd_detail/dlopen/cholmod.hpp
index 0864172..342ed4f 100644
--- a/library/src/amd_detail/dlopen/cholmod.hpp
+++ b/library/src/amd_detail/dlopen/cholmod.hpp
@@ -24,7 +24,7 @@
 #include "lib_macros.hpp"
 
 #ifdef HAVE_ROCSPARSE
-#include <suitesparse/cholmod.h>
+#include <cholmod.h>
 #else
 
 // constants
-- 
2.43.0

